compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -653,15 +653,18 @@ object Denotations {
       case ExprType(rtp1) =>
         tp2 match {
           case ExprType(rtp2) => ExprType(rtp1 & rtp2)
-          case _ => rtp1 & tp2
+          case _ => infoMeet(rtp1, tp2, sym1, sym2, safeIntersection)
```

Comment:
Can this ever actually succeed ? I assume `tp2` is always go to be a method type or a poly type so we'll end up with a mergeConflict. If so we might as well directly return with a mergeConflict here:
```suggestion
          case _ => mergeConflict(sym1, sym2, tp1, tp2)
```
```

Reply:
Ah, I guess this would actually suceed if we're merging a `val` and a nilary `def`, nevermind then.